realm/realm-library/src/main/cpp/io_realm_ObjectStoreUserStore.cpp

```diff
@@ -0,0 +1,138 @@
+    } else if (all_users.size() < 1) {
+        std::runtime_error("no user logged-in yet.");//TODO user appropriate exception
+    } else {
+        return all_users.front();
+    }
+}
+
+JNIEXPORT jstring JNICALL Java_io_realm_ObjectStoreUserStore_getCurrentUser
+        (JNIEnv *env, jclass) {
+    TR_ENTER()
+    const std::shared_ptr<SyncUser>& user = currentUserOrThrow();
+    if (user->state() == SyncUser::State::Active) {
+        return env->NewStringUTF(user->refresh_token().c_str());
+    } else {
+        return NULL;
+    }
+}
+
+/*
+ * Class:     io_realm_ObjectStoreSyncManager
+ * Method:    updateOrCreateUser
+ * Signature: (Ljava/lang/String;Ljava/lang/String;)V
+ */
+JNIEXPORT void JNICALL Java_io_realm_ObjectStoreUserStore_updateOrCreateUser
+        (JNIEnv *env, jclass, jstring identity, jstring jsonToken, jstring url) {
+    TR_ENTER()
+    try {
+        JStringAccessor user_identity(env, identity); // throws
+        JStringAccessor user_json_token(env, jsonToken); // throws
+        JStringAccessor auth_url(env, url); // throws
+
+        SyncManager::shared().get_user(user_identity, user_json_token, util::Optional<std::string>(auth_url));
+    } CATCH_STD()
+}
+
+/*
+ * Class:     io_realm_ObjectStoreSyncManager
+ * Method:    logoutCurrentUser
+ * Signature: ()V
+ */
+JNIEXPORT void JNICALL Java_io_realm_ObjectStoreUserStore_logoutCurrentUser
+        (JNIEnv *, jclass) {
+    TR_ENTER()
+    const std::shared_ptr<SyncUser>& user = currentUserOrThrow();
+    user->log_out();
+}
+
+
+/*
+ * Class:     io_realm_ObjectStoreSyncManager
+ * Method:    configureMetaDataSystem
+ * Signature: (Ljava/lang/String;[B)V
+ */
+JNIEXPORT void JNICALL Java_io_realm_ObjectStoreUserStore_configureMetaDataSystem
+        (JNIEnv *env, jclass, jstring baseFile, jbyteArray aesKey) {
+    TR_ENTER()
+    try {
+        JStringAccessor base_file_path(env, baseFile); // throws
+        SyncManager::shared().configure_file_system(base_file_path, SyncManager::MetadataMode::NoEncryption);
+        //TODO use encryption mode with the provided key
+        //     JniByteArray key_array(env, aesKey);
+    } CATCH_STD()
+
+}
+
+/*
+ * Class:     io_realm_ObjectStoreSyncManager
+ * Method:    getAllUsers
+ * Signature: ()[Ljava/lang/String;
+ */
+JNIEXPORT jobjectArray JNICALL Java_io_realm_ObjectStoreUserStore_getAllUsers
+        (JNIEnv *env, jclass) {
+    TR_ENTER()
+    std::vector<std::shared_ptr<SyncUser>> all_users = SyncManager::shared().all_users();
+    if (!all_users.empty()) {
+        std::vector<jobject> valid_users;
+        for (auto& user : all_users) {
+            if ((*user).state() == SyncUser::State::Active) {
+                valid_users.push_back(env->NewStringUTF((*user).refresh_token().c_str()));
+            }
+        }
+
+        std::vector<jobject>::size_type size_valid_users = valid_users.size();
+        jclass stringClass = env->FindClass("java/lang/String");
```

Comment:
Please use `GetClass()` in `util.hpp`. Or look it up when the `.so` is loaded (see `JNI_OnLoad` in `io_realm_internal_Util.cpp`)